pkg/concurrency/stopper_test.go

```diff
@@ -0,0 +1,126 @@
+package concurrency
+
+import (
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/suite"
+)
+
+type StopperTestSuite struct {
+	suite.Suite
+}
+
+func TestStopper(t *testing.T) {
+	suite.Run(t, &StopperTestSuite{})
+}
+
+// TestCommonCase demonstrates how to use Stopper for implementing a gracefully stoppable goroutine.
+func (s *StopperTestSuite) TestCommonCase() {
+	stopper := NewStopper()
+
+	s1 := NewSignal()
+	s2 := NewSignal()
```

Comment:
I guess some readers of this code may ask themselves how to understand s1 and s2.
It s1 a signal to start the work in the goroutine and s2 a signal to mark the end of that work? Maybe we could rename s1 and s2 so that their roles are easier to understand?

Reply:
I thought about this but can't find good names. Also, the code is small and localized that just reading it explains what `s1` and `s2` do. I think, in this case it's like explaining a joke. Don't want to spoil the joy :-)